### PR TITLE
python3Packages.{altgraph,pyinstaller-hooks-contrib,macholib,PyInstaller,keyboard}: init at {0.17,2021.2,1.14,4.3,0.13.5}

### PR DESCRIPTION
**Pull request by Kranzes**
###### Motivation for this change
Packaged new Python packages which i will need in order to package a certain program that i plan on packaging.
Also added myself as a maintainer since its needed in order to add packages to nixpkgs.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by fufexan** [comment]
Looks fine apart from the formatting checks.

**Comment by Kranzes** [comment]
> Looks fine apart from the formatting checks.

Will push formatting fixes in a bit.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -5333,6 +5339,10 @@ in {
 
   pyisy = callPackage ../development/python-modules/pyisy { };
 
+  PyInstaller = callPackage ../development/python-modules/PyInstaller { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:5342]:
```suggestion
  pyinstaller = callPackage ../development/python-modules/pyinstaller { };
```

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -3854,6 +3856,8 @@ in {
   keras-preprocessing = callPackage ../development/python-modules/keras-preprocessing { };
 
   kerberos = callPackage ../development/python-modules/kerberos { };
+
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3859]:
```suggestion

```

File: pkgs/development/python-modules/keyboard/default.nix [hunk 3]
```diff
@@ -0,0 +1,22 @@
+{ lib, fetchFromGitHub, python3Packages }:
+
+python3Packages.buildPythonPackage rec {
+  pname = "keyboard";
+  version = "0.13.5";
+
+  src = fetchFromGitHub {
+    owner = "boppreh";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/keyboard/default.nix:8]:
```suggestion
    owner = "boppreh";
```

File: pkgs/development/python-modules/pyinstaller-hooks-contrib/default.nix [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+python3Packages.buildPythonPackage rec {
+  pname = "pyinstaller-hooks-contrib";
+  version = "2021.2";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-f10GibMNowkhSfxTaoNalARayMnw5t+yOsFxiQ9eqPI=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyinstaller-hooks-contrib/default.nix:11]: pythonImportsCheck and checkPhase?

File: pkgs/development/python-modules/macholib/default.nix [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  version = "1.14";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-DENryEfnsdm9oFYDUb9218r5MPtYWoKNE2CIOe9CxDI=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [ altgraph ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/macholib/default.nix:14]: pythonImportsCheck is required when disabling tests and also add a comment why you disabled tests.

File: pkgs/development/python-modules/keyboard/default.nix [hunk 6]
```diff
@@ -0,0 +1,22 @@
+{ lib, fetchFromGitHub, python3Packages }:
+
+python3Packages.buildPythonPackage rec {
+  pname = "keyboard";
+  version = "0.13.5";
+
+  src = fetchFromGitHub {
+    owner = "boppreh"; 
+    repo = "${pname}";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/keyboard/default.nix:9]:
```suggestion
    repo = pname;
```

File: pkgs/development/python-modules/keyboard/default.nix [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-U4GWhPp28azBE3Jn9xpLxudOKx0PjnYO77EM2HsJ9lM=";
+  };
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/keyboard/default.nix:14]: pythonImportsCheck is required when disabling tests and also add a comment why you disabled tests.

File: pkgs/development/python-modules/altgraph/default.nix [hunk 8]
```diff
@@ -0,0 +1,18 @@
...
+
+python3Packages.buildPythonPackage rec {
+  pname = "altgraph";
+  version = "0.17";
+
+  src = fetchPypi {
+    inherit pname version;
+    sha256 = "sha256-HwWkcSJUL5cCjK94d1oJX75qJpm1CJ3oR361gxZ9aao=";
+  };
+
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/altgraph/default.nix:11]: pythonImportsCheck and checkPhase?

File: pkgs/development/python-modules/PyInstaller/default.nix [hunk 9]
```diff
@@ -0,0 +1,27 @@
...
+    sha256 = "sha256-Xs+LvCMNcpinluUrt0W5Xu4Sh40UHxZFYSyZJG7NI/I=";
+  };
+
+  propagatedBuildInputs = with python3Packages; [
+    altgraph
+    pyinstaller-hooks-contrib
+    macholib
+  ];
+
+  doCheck = false;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/PyInstaller/default.nix:19]: pythonImportsCheck is required when disabling tests and also add a comment why you disabled tests.

File: pkgs/development/python-modules/PyInstaller/default.nix [hunk 10]
```diff
@@ -0,0 +1,27 @@
...
+
+  meta = with lib; {
+    description = "Freeze (package) Python programs into stand-alone executables";
+    homepage = "http://www.pyinstaller.org";
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/PyInstaller/default.nix:23]:
```suggestion
    homepage = "https://www.pyinstaller.org/";
```

File: maintainers/maintainer-list.nix [hunk 11]
```diff
@@ -11980,4 +11980,10 @@
     github = "zupo";
     githubId = 311580;
   };
+  kranzes = {
```
Comment (SuperSandro2000) [inline maintainers/maintainer-list.nix:11983]: Please put this into a separate commit with the message ``maintainers: add NAME`` and sort it alphabetically.

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Using python3Packages in pythonPackages is forbidden. Remove it from all inputs and replace it with the subpackages and functions you need.

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 129344` at 975a9709 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/0ac99edeb759cab878d0dd1ad7412541">python38Packages.PyInstaller</a> (python39Packages.PyInstaller)</li>
  </ul>
</details>
<details>
  <summary>4 packages built successfully:</summary>
  <ul>
    <li>python38Packages.altgraph (python39Packages.altgraph)</li>
    <li>python38Packages.keyboard (python39Packages.keyboard)</li>
    <li>python38Packages.macholib (python39Packages.macholib)</li>
    <li>python38Packages.pyinstaller-hooks-contrib (python39Packages.pyinstaller-hooks-contrib)</li>
  </ul>
</details>
<details>
  <summary>4 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/no-python-tests.md">no-python-tests</a>

Test runner could not discover any test cases: ‘Ran 0 tests in 0.000s’
Near pkgs/development/python-modules/pyinstaller-hooks-contrib/default.nix:13:0:
```
   |
13 |     description = "Community maintained hooks for PyInstaller";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/keyboard/default.nix:17:0:
```
   |
17 |     description = "Hook and simulate keyboard events";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/macholib/default.nix:17:0:
```
   |
17 |     description = "Mach-O header analysis and editing";
   | ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/python-include-tests.md">python-include-tests</a>

Consider adding a `checkPhase` for tests, or if not feasible, `pythonImportsCheck`.

Near pkgs/development/python-modules/PyInstaller/default.nix:22:0:
```
   |
22 |     description = "Freeze (package) Python programs into stand-alone executables";
   | ^
```
</li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

File: pkgs/top-level/python-packages.nix [hunk 12]
```diff
@@ -3854,6 +3856,7 @@ in {
   keras-preprocessing = callPackage ../development/python-modules/keras-preprocessing { };
 
   kerberos = callPackage ../development/python-modules/kerberos { };
+  keyboard = callPackage ../development/python-modules/keyboard { };
```
Comment (SuperSandro2000) [inline pkgs/top-level/python-packages.nix:3859]:
```suggestion

  keyboard = callPackage ../development/python-modules/keyboard { };
```

File: pkgs/development/python-modules/pyinstaller/default.nix [hunk 13]
```diff
@@ -0,0 +1,27 @@
+{ lib, fetchPypi, python3Packages }:
+
+python3Packages.buildPythonPackage rec {
+  pname = "PyInstaller";
+  version = "4.3";
+
+  src = fetchPypi {
+    pname = "pyinstaller";
+    inherit version;
```
Comment (SuperSandro2000) [inline pkgs/development/python-modules/pyinstaller/default.nix:9]:
```suggestion
  pname = "pyinstaller";
  version = "4.3";

  src = fetchPypi {
    inherit pname version;
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]
Please cleanup the history that it follow the contributing guide.

and all packages still do not have checkInputs/checkPhases and pythonImportsChecks. Please give me a ping when you are done.

**Comment by Kranzes** [comment]
Will do, just don't have the time to fix it all at once.

**Comment by Kranzes** [comment]
Closing for now due to a lot of complexity between these modules which i still need figure out.

**Comment by Eveeifyeve** [comment]
I have publised the macholib package.
